packages/cli/src/lib/bundler/discover.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstagePackageJson } from '@backstage/cli-node';
+import { Config } from '@backstage/config';
+import fs from 'fs-extra';
+import path from 'path';
+import chokidar from 'chokidar';
+
+import { paths as cliPaths } from '../../lib/paths';
+import { BundlingPathsOptions, resolveBundlingPaths } from './paths';
+
+type Options = { config: Config; watch: () => void } & BundlingPathsOptions;
+
+export async function buildDetectedPlugins(options: Options) {
+  const { entry, targetDir } = options;
+  const { targetPackageJson } = resolveBundlingPaths({ entry, targetDir });
+
+  if (!!options.watch) {
+    const watcher = chokidar.watch(targetPackageJson);
+
+    watcher.on('change', async () => {
+      await writeDetectedPluginsModule(options);
+      options.watch();
```

Comment:
```suggestion
      options.onChange();
```

Should we consider only invoking the `writeDetectedPluginsModule` and onChange callback when only dependencies are changed?